### PR TITLE
Merge Ng-For-Static-Pages into beta-main

### PR DESCRIPTION
**Pull request by DabeerAzeez**
- Convert About and FAQ pages to use .json's and *ngFor's to clean up the HTML code

### CODE REVIEW

**Review by ricetech — CHANGES_REQUESTED** [review]
You are incorrectly displaying properties that don't exist (websites/instagrams for people that don't have them)

**Comment by DabeerAzeez** [comment]
@ricetech I thought we wanted that for consistency? Or at least having all the social links in the same order. I remember we mentioned doing it that way before but I just ended up only putting the ones that were there. @janakitti @TommyHuang-dev what do you guys think?

Not hard to change, just an `*ngIf`.

**Comment by DabeerAzeez** [comment]
@TommyHuang-dev has agreed with you, @ricetech. I will make the changes.

**Comment by janakitti** [comment]
Yeah I guess I agree with @ricetech too since it doesn't make sense to have links to nowhere. You can just have an ngIf inside the ngFor to check if a field is nonexistent.

**Comment by ricetech** [comment]
> @ricetech I thought we wanted that for consistency? Or at least having all the social links in the same order. I remember we mentioned doing it that way before but I just ended up only putting the ones that were there. @janakitti @TommyHuang-dev what do you guys think?
> 
> Not hard to change, just an `*ngIf`.

They should be in the same order for consistency, yes, but that doesn't involve showing fields that don't exist - it just makes it look like we forgot to ask the person for their social media handles.

File: src/app/pages/about/about.component.html [hunk 1]
```diff
@@ -25,251 +25,48 @@ <h1 class="mb-3" id="title">About Us</h1>
...
+          <p>
+            <a href="{{contributor.sharingLink}}" target="_blank">Click here to sign {{contributor.name}}'s virtrolio.</a>
+          </p>
+        </div>
+
+        <!-- Each Contact Info. -->
+        <div class="col-md-3 order-md-1 vertical-center">
+          <p><strong>Contact {{contributor.name}}</strong></p>
+          <ul>
+            <li *ngIf="contributor.social.github">Github:
```
Comment (janakitti) [inline src/app/pages/about/about.component.html]:
```suggestion
            <li *ngIf="contributor.social.github">GitHub:
```

**Review by janakitti — APPROVED** [review]

**Review by LillianHuang-dev — APPROVED** [review]

File: src/app/pages/about/about.component.html [hunk 2]
```diff
@@ -25,251 +25,48 @@ <h1 class="mb-3" id="title">About Us</h1>
...
-      </div>
-
+    <!-- Each Row-->
+    <div *ngFor="let contributor of contributors">
+      <div class="row d-flex my-2">
+
+        <!-- Each Image -->
+        <div class="col-md-2 order-md-2 image-container vertical-center">
+          <img alt="Picture of {{contributor.name}}" class="img-fluid regular-picture" src="{{ contributor.images.regular }}">
+          <img alt="Silly picture of {{contributor.name}}" class="img-fluid silly-picture" src="{{ contributor.images.silly }}">
```
Comment (ricetech) [inline src/app/pages/about/about.component.html]: You can't add spaces in one tag and leave them out in the next xD
```suggestion
          <img alt="Picture of {{ contributor.name }}" class="img-fluid regular-picture" src="{{ contributor.images.regular }}">
          <img alt="Silly picture of {{ contributor.name }}" class="img-fluid silly-picture" src="{{ contributor.images.silly }}">
```

File: src/app/pages/about/about.component.html [hunk 3]
```diff
@@ -25,251 +25,48 @@ <h1 class="mb-3" id="title">About Us</h1>
...
+        </div>
+
+        <!-- Each Blurb -->
+        <div class="col-md-7 order-md-3 mt-3 mt-md-0 vertical-center">
+          <p class="text-justify" [innerHTML]="contributor.blurb"></p>
+          <p>
+            <a href="{{contributor.sharingLink}}" target="_blank">Click here to sign {{contributor.name}}'s virtrolio.</a>
```
Comment (ricetech) [inline src/app/pages/about/about.component.html]: spaces
```suggestion
            <a href="{{ contributor.sharingLink }}" target="_blank">Click here to sign {{ contributor.name }}'s virtrolio.</a>
```

File: src/app/pages/about/about.component.html [hunk 4]
```diff
@@ -25,251 +25,48 @@ <h1 class="mb-3" id="title">About Us</h1>
...
+            </li>
+            <li *ngIf="contributor.social.instagram">Instagram:
+              <a href="https://www.instagram.com/{{contributor.social.instagram}}"
+                 target="_blank" *ngIf="contributor.social.instagram">@{{contributor.social.instagram}}</a>
+            </li>
+            <li *ngIf="contributor.social.linkedIn">LinkedIn:
+              <a href="{{contributor.social.linkedIn}}" target="_blank">{{contributor.name}}</a>
+            </li>
+            <li *ngIf="contributor.social.website">Website:
+              <a href="https://{{contributor.social.website}}" target="_blank">{{contributor.social.website}}</a>
```
Comment (ricetech) [inline src/app/pages/about/about.component.html]:
```suggestion
          <p><strong>Contact {{ contributor.name }}</strong></p>
          <ul>
            <li *ngIf="contributor.social.github">GitHub:
              <a href="https://github.com/{{ contributor.social.github }}"
                 target="_blank">{{ contributor.social.github }}</a>
            </li>
            <li *ngIf="contributor.social.instagram">Instagram:
              <a href="https://www.instagram.com/{{ contributor.social.instagram }}"
                 target="_blank" *ngIf="contributor.social.instagram">@{{ contributor.social.instagram }}</a>
            </li>
            <li *ngIf="contributor.social.linkedIn">LinkedIn:
              <a href="{{ contributor.social.linkedIn }}" target="_blank">{{ contributor.name }}</a>
            </li>
            <li *ngIf="contributor.social.website">Website:
              <a href="https://{{ contributor.social.website }}" target="_blank">{{ contributor.social.website }}</a>
```

File: src/app/pages/faq/faq.component.html [hunk 5]
```diff
@@ -2,287 +2,29 @@
   <h1 class="mt-3 mb-3" id="title">Frequently Asked Questions</h1>
   <br/>
 
-  <!-- Section 1: How to use a virtrolio? -->
-  <h2>Signing and Sharing your virtrolio</h2>
-  <div class="accordion" id="faq-s1">
+  <div class="accordion mb-3" id="section-header{{s}}" *ngFor="let section of faqContent; let s = index">
+    <!-- Each section-->
+    <h2>{{section.sectionTitle}}</h2>
```
Comment (ricetech) [inline src/app/pages/faq/faq.component.html]:
```suggestion
    <h2>{{ section.sectionTitle }}</h2>
```

File: src/app/pages/faq/faq.component.html [hunk 6]
```diff
@@ -2,287 +2,29 @@
...
 
-    <!-- Question -->
-    <div class="card">
-      <div class="card-header" id="s1q1h">
+    <!-- Each question per section -->
+    <div class="card" *ngFor="let question of section.questions; let q = index">
+      <div class="card-header" id="question{{s}}-{{q}}">
```
Comment (ricetech) [inline src/app/pages/faq/faq.component.html]:
```suggestion
      <div class="card-header" id="question{{ s }}-{{ q }}">
```

File: src/app/pages/faq/faq.component.html [hunk 7]
```diff
@@ -2,287 +2,29 @@
...
         <h2 class="mb-0">
-          <button aria-controls="s1q1c" aria-expanded="false" class="btn btn-link" data-target="#s1q1c"
+          <button [attr.aria-controls]="'answer'+s+'-'+q" aria-expanded="false" class="btn btn-link" [attr.data-target]="'#answer'+s+'-'+q"
                   data-toggle="collapse" type="button">
-            How do I sign someone's virtrolio?
+            {{question.question}}
```
Comment (ricetech) [inline src/app/pages/faq/faq.component.html]:
```suggestion
            {{ question.question }}
```

File: src/app/pages/faq/faq.component.html [hunk 8]
```diff
@@ -2,287 +2,29 @@
...
-
-      <!-- Answer -->
-      <div aria-labelledby="s1q7h" class="collapse" data-parent="#faq-s1" id="s1q7c">
-        <div class="card-body card-align">
-          You can generate a new sharing link if people are sending you messages you do not approve of using your old
-          one. This helps prevent spammers and bullies from clogging up your virtrolio.
-          Only people with your <em>newest</em> sharing link can sign your virtrolio.
-        </div>
+      <!-- Each answer per question -->
+      <div aria-labelledby="question" class="collapse" [attr.data-parent]="'#section-header'+s" id="answer{{s}}-{{q}}">
```
Comment (ricetech) [inline src/app/pages/faq/faq.component.html]:
```suggestion
      <div aria-labelledby="question" class="collapse" [attr.data-parent]="'#section-header'+s" id="answer{{ s }}-{{ q }}">
```

**Review by ricetech — CHANGES_REQUESTED** [review]

**Review by ricetech — APPROVED** [review]

**Pull request merged**